pip/resolve.py

```diff
@@ -123,11 +123,26 @@ class Resolver(object):
     the requested operation without breaking the requirements of any package.
     """
 
-    def __init__(self, upgrade_strategy, finder):
+    _allowed_strategies = ["eager", "only-if-needed", "to-satisfy-only"]
+
+    def __init__(self, session, finder, use_user_site,
+                 ignore_dependencies, ignore_installed, ignore_requires_python,
+                 force_reinstall, isolated, upgrade_strategy):
         super(Resolver, self).__init__()
-        assert upgrade_strategy in ["eager", "only-if-needed", "not-allowed"]
-        self.upgrade_strategy = upgrade_strategy
+        assert upgrade_strategy in self._allowed_strategies
+
         self.finder = finder
+        self.session = session
+
+        self.require_hashes = None  # This is set in resolve
```

Comment:
Another good use of comments to indicate where this is going to end up being set.